### ISSUE TITLE
Style not apply on Inherited component

### ISSUE THREAD

**Issue by steam3d** [microsoft/microsoft-ui-xaml#8869]
### Describe the bug

When Inherited from any component style of WinUI 2.X is not apply.

### Steps to reproduce the bug

cs
```
public partial class SettingSlider : Slider
{               
}
```

xaml
```
<local:SettingSlider />
```

### Expected behavior

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/30986874/15a15ffb-778a-4b58-8683-5b3add8c905f)

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/30986874/f4e8dff4-b051-4a2e-88a2-de28516cde4b)

### NuGet package version

WinUI 2 - Microsoft.UI.Xaml 2.8.2

**Comment by zadjii-msft** [comment]
I'm pretty sure this is "by-design" with Controlsv2, but I don't know if there's a thread for that or not. 

In the Terminal, we worked around this by manually setting the `Style` to the default style for that control. (for us it was a ContentDialog, but the same principal applies):

https://github.com/microsoft/terminal/blob/f494d68f11b457bbbdab4773cc605d6962fb6139/src/cascadia/TerminalApp/AboutDialog.xaml#L15

**Comment by zadjii-msft** [comment]
Actually this commit may be more relevant:
https://github.com/microsoft/terminal/commit/86d33a970ae33586c45314751c2a4834e85426ac

**Comment by steam3d** [comment]
Yep. I already figure out the same way like you said, but it will be better if Microsoft somewhere mention this.
